### PR TITLE
@graph-paper/datagraphic: exits extents for categorical plots; cleans up mouse position logic

### LINKED ISSUE #77: simplify createMouseStore in DataGraphic.svelte
The mouseover functionality is needlessly complex and feels like a misuse of stores.

### LINKED ISSUE #97: extents calculation must accomodate scalePoint and scaleBand
The extents code in DataGraphic primarily benefits `linear` and `time`. They need to accomodate `scalePoint` and `scaleBand` as well.

### LINKED ISSUE #99: rename hoverValue to mousePosition everywhere
This is more important API work that needs to be done before end of the quarter. `hoverValue` is used pretty much everywhere and needs to be `mousePosition` to match the API needs.

### PR DESCRIPTION
**Pull request by hamilton**
closes #97; closes #77; closes #99 

What's left for this work:

- [x] remove the convoluated mouseover store
- [x] updates slot props / availability of `hoverValue` -> `mousePosition`
- [x] update changelog
- [x] update readme

### CODE REVIEW

**Comment by hamilton** [comment]
@openjck @wlach tagged the two of you for a quick review, if you have a chance this weekend. No obligation to review over a weekend of course – just a courtesy. I'm going to let this PR sit for a day and then revisit tomorrow.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 - Change `xDomainMin` / `xDomainMax` & y equivalents to `xMin` / `xMax` etc. ([#95](https://github.com/graph-paper-org/graph-paper/pull/95))
 - Refactor width / height of `DataGraphic` component to reactively update ([#83](https://github.com/graph-paper-org/graph-paper/pull/82))
 - Change `borderThickness` to `borderSize` in `DataGraphic` ([#96](https://github.com/graph-paper-org/graph-paper/pull/96))
+- Refactor mouse position calculations ([#98](https://github.com/graph-paper-org/graph-paper/pull/98))
+- Renames `hoverValue` to `mousePosition` in `DataGraphic` and throughout all code examples ([#98](https://github.com/graph-paper-org/graph-paper/pull/98))
```
Comment (hamilton) [inline CHANGELOG.md:9]:
```suggestion
- Rename `hoverValue` to `mousePosition` in `DataGraphic` and throughout all code examples ([#98](https://github.com/graph-paper-org/graph-paper/pull/98))
```

File: packages/datagraphic/DataGraphic.svelte [hunk 2]
```diff
@@ -251,13 +252,21 @@
...
   const initialXDomain = initializeDomain(xType);
   const initialYDomain = initializeDomain(yType);
-  let internalXDomain = tweened(initializeDomain(xType), internalXDomainTween);
-  let internalYDomain = tweened(initializeDomain(yType), internalYDomainTween);
+  let internalXDomain = isContinuous(xType)
+    ? tweened(initializeDomain(xType), internalXDomainTween)
+    : writable(initializeDomain(xType));
+  let internalYDomain = isContinuous(yType)
+    ? tweened(initializeDomain(yType), internalYDomainTween)
+    : writable(initializeDomain(yType));
```
Comment (wlach) [inline packages/datagraphic/DataGraphic.svelte:269]: Can you somehow refactor this code so we're not duplicating the same logic between X and Y? Maybe define some kind of initialization helper function and call it twice?

File: packages/datagraphic/mouse-position.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { range, bisect } from "d3-array";
+
+export function toSVGPixelCoordinates(x, y, svgElement) {
+  const pt = svgElement.createSVGPoint();
+  pt.x = x;
+  pt.y = y;
```
Comment (wlach) [inline packages/datagraphic/mouse-position.js:6]:
```suggestion
  const pt = {
    ...svgElement.createSVGPoint()
    x,
    y
  };
```

**Comment by hamilton** [comment]
Thanks for the review @wlach!

**Comment by openjck** [comment]
Aw dang, got in before I had time to review. Thanks Will!

**Pull request merged**